docs/source/design/python_imports.rst

```diff
@@ -0,0 +1,61 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_python_imports:
+
+************************
+Python import statements
+************************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current (albeit incomplete) implementation.
+
+The following rules apply to any ``import`` statement in the code base:
+
+- All imports *must* be absolute.
+
+- Imports *must* be placed at the top of a source file, unless there is a
+  specific reason not to do so (e.g., delayed import due to performance
+  concerns, circular dependencies). If such a reason exists, it *must*
+  should be documented by a comment at the import statement.
+
+- There *must* be no more than one import per line.
+
+- Multiple individual imports from a single module *must* follow the pattern::
+
+      from <module> import (
+          symbol1,
+          symbol2,
+      )
+
+  Individual imported symbols *should* be sorted alphabetically. The last symbol
+  line *should* end with a comma.
+
+- Imports from packages and modules *may* be grouped in categories like
+
+  - Standard library packages
+
+  - 3rd-party packages
+
+  - Datalad core
+
+  - Datalad extensions
```

Comment:
we should check how things would look like with https://github.com/PyCQA/isort 

```
  - Datalad extensions
```
is relevant only within extensions

Echoing above, may be 
```suggestion
  - Datalad core (absolute imports)

  - Datalad extensions
  
  - DataLad core ("local" relative imports)
```

Reply:
```suggestion
  - Datalad core (absolute imports)

  - Datalad extensions
  
  - Datalad core ("local" relative imports)
  
  Sorting imports can be aided by https://github.com/PyCQA/isort (e.g. `python -m isort -m3 -fgw 2 -tc <filename>`).
```